src/content/app/species-selector/SpeciesSelector.tsx

```diff
@@ -30,33 +26,20 @@ import SpeciesSelectorGeneSearchView from './views/species-selector-gene-search-
 import styles from './SpeciesSelector.module.css';
 
 const SpeciesSelector = () => {
-  const modalView = useAppSelector(getSpeciesSelectorModalView);
-  const dispatch = useAppDispatch();
-
-  const shouldShowResultsView = [
-    'species-search',
-    'popular-species-genomes'
-  ].includes(modalView || '');
-
-  useEffect(() => {
-    return () => {
-      // close the modal view when leaving Species Selector
-      dispatch(setModalView(null));
-    };
-  }, []);
-
-  const appBar = shouldShowResultsView ? (
-    <SpeciesSearchResultsModalAppBar />
-  ) : (
-    <SpeciesSelectorAppBar />
+  const appBar = (
+    <Routes>
+      <Route index element={<SpeciesSelectorAppBar />} />
+      <Route path="/search" element={<SpeciesSearchResultsModalAppBar />} />
+      <Route path="/search/gene" element={<SpeciesSelectorAppBar />} />
+    </Routes>
   );
 
-  const body = shouldShowResultsView ? (
-    <SpeciesSelectorResultsView />
-  ) : modalView === 'gene-search' ? (
-    <SpeciesSelectorGeneSearchView />
-  ) : (
-    <SpeciesSelectorMainView />
+  const body = (
+    <Routes>
+      <Route index element={<SpeciesSelectorMainView />} />
+      <Route path="/search" element={<SpeciesSelectorResultsView />} />
+      <Route path="/search/gene" element={<SpeciesSelectorGeneSearchView />} />
```

Comment:
One thing to verify is that we expect the gene search view to be accessible only from the species selector. Technically, there is nothing species selector-ish in gene search view. Ideally, I would have expected it to go inside of `/search` app.